ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/RecoveryTest.java

```diff
@@ -625,6 +629,51 @@ public void shouldRecoverLogWithTerminateAfterDrop() {
     shouldRecover(commands);
   }
 
+  @Test
+  public void shouldNotRecoverIfStatementIsNoLongerValid() {
+    // Given:
+    topicClient.preconditionTopicExists("B");
+
+    server1.submitCommands(
+        "CREATE STREAM A (COLUMN STRING) WITH (KAFKA_TOPIC='A', VALUE_FORMAT='JSON');",
+        "CREATE STREAM B (COLUMN STRING) WITH (KAFKA_TOPIC='B', VALUE_FORMAT='JSON');"
+    );
+
+    // When:
+    // after deleting topic "A", the first statement will no longer be able to run,
+    // which will cause the other statements to fail
```

Comment:
NOTE: this might not be acceptable behavior - it is likely that for a long-running server, some queries will eventually be dropped and their topics deleted. We should be able to recover from this scenario.